### PR TITLE
74038 - more tests on create/update step handlers

### PR DESCRIPTION
**Pull request by llykkelarsen**

### CODE REVIEW

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs [hunk 1]
```diff
@@ -90,6 +117,34 @@ public async Task HandlingCreateStepCommand_ShouldSave()
...
+            await _dut.Handle(_command, default);
+
+            // Assert
+            _responsibleRepositoryMock.Verify(r => r.Add(It.IsAny<Responsible>()), Times.Once);
+            UnitOfWorkMock.Verify(r => r.SaveChangesAsync(default), Times.Exactly(2));
+            Assert.AreEqual(_addedResponsible.Code, ResponsibleCode);
+        }
+
+        [TestMethod]
+        public async Task HandlingCreateStepCommand_ShouldNotAddResponsibleToStep_WhenResponsibleAlreadyExists()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs:138]: HandlingCreateStepCommand_ShouldNotAddResponsibleToRepository_WhenResponsibleAlreadyExists

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs [hunk 2]
```diff
@@ -90,6 +117,34 @@ public async Task HandlingCreateStepCommand_ShouldSave()
...
+        {
+            Assert.AreEqual(0, _journey.Steps.Count);
+
+            // Act
+            await _dut.Handle(_command, default);
+
+            // Assert
+            _responsibleRepositoryMock.Verify(r => r.Add(It.IsAny<Responsible>()), Times.Never);
+            Assert.AreEqual(ResponsibleId, _journey.Steps.First().ResponsibleId);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs:147]: I don't understand this Assert.AreEqual here.
Should be Assert.IsNull(_addedResponsible) i think

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs [hunk 3]
```diff
@@ -90,6 +117,34 @@ public async Task HandlingCreateStepCommand_ShouldSave()
             UnitOfWorkMock.Verify(u => u.SaveChangesAsync(default), Times.Once);
         }
 
-        //TODO: Add more tests
+        [TestMethod]
+        public async Task HandlingCreateStepCommand_ShouldAddResponsibleToStep_WhenResponsibleNotExists()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs:121]: HandlingCreateStepCommand_ShouldAddResponsibleToRepository_WhenResponsibleNotExists

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs [hunk 4]
```diff
@@ -143,5 +161,38 @@ public async Task HandlingUpdateStepCommand_ShouldSave()
             // Assert
             UnitOfWorkMock.Verify(u => u.SaveChangesAsync(default), Times.Once);
         }
+
+        [TestMethod]
+        public async Task HandlingUpdateStepCommand_ShouldAddResponsibleToStep_WhenResponsibleNotExists()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs:166]: same as above

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs [hunk 5]
```diff
@@ -143,5 +161,38 @@ public async Task HandlingUpdateStepCommand_ShouldSave()
...
+
+            // Assert
+            _responsibleRepositoryMock.Verify(r => r.Add(It.IsAny<Responsible>()), Times.Once);
+            UnitOfWorkMock.Verify(r => r.SaveChangesAsync(default), Times.Exactly(2));
+            Assert.AreEqual(_addedResponsible.Code, _responsibleCode);
+
+        }
+
+        [TestMethod]
+        public async Task HandlingUpdateStepCommand_ShouldNotAddResponsibleToStep_WhenResponsibleAlreadyExists()
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs:184]: same as above

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs [hunk 6]
```diff
@@ -143,5 +161,38 @@ public async Task HandlingUpdateStepCommand_ShouldSave()
...
+            // Arrange
+            Assert.AreEqual(1, _journey.Steps.Count);
+            Assert.AreEqual(_responsibleId, _journey.Steps.First().ResponsibleId);
+
+            // Act
+            await _dut.Handle(_commandWithResponsible2, default);
+
+            // Assert
+            _responsibleRepositoryMock.Verify(r => r.Add(It.IsAny<Responsible>()), Times.Never);
+            Assert.AreEqual(_responsibleId2, _journey.Steps.First().ResponsibleId);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/UpdateStep/UpdateStepCommandHandlerTests.cs:195]: same as above

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs (hunk 2 above)
Comment (llykkelarsen) [reply src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs:147]: The thought was to check if although the Add method has not been called, the existing responsible has been added?

Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.Command.Tests/JourneyCommands/CreateStep/CreateStepCommandHandlerTests.cs:147]: I C

**Pull request merged**